### ISSUE TITLE
Unexpected lucene-text-search results

### ISSUE THREAD

**Issue by tkovis** [xtdb/xtdb#1594]
If I transact new documents (eg. test1) for debugging purposes, I can find them with both `q` and `lucene-text-search`. But if I query documents that have been transacted as a part of an etl job from our old database, I can only with them with `q`. I now deleted tx-log, document-store and indices and I'm trying to redo the etl job to see if the problems persists.

```clojure
(crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :user/first-name])]
    :where '[[?e :user/first-name "James"]]
    :limit 1})
  (comment
    [[{:crux.db/id "test1"
       :user/first-name
       "James"}]])

  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :user/first-name]) ?s]
    :where '[[(lucene-text-search "(user\\/first-name:James)") [[?e ?s]]]]
    :limit 1})
  (comment
    #{[{:crux.db/id "test1"
        :user/first-name
        "James"}
       0.08287343382835388]})

  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name])]
    :where '[[?e :person/last-name "taateliha"]]
    :limit 1})
  (comment
    [[{:crux.db/id
       #uuid
        "003bc142-7e4e-4221-b8c4-4730a70f1da1"
       :person/last-name
       "taateliha"}]])

  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [*]) ?s]
    :where '[[(lucene-text-search "(person\\/last-name:taateliha)") [[?e ?s]]]]})
  (comment
    #{})
```

**Comment by tkovis** [comment]
Yeah it persists
```clojure
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name])]
    :where '[[?e :person/last-name "taateliha"]]
    :limit 1})
  (comment
    [[{:crux.db/id
       #uuid
        "0021cb70-4e9b-40fe-b3fc-d92ef16b02a7"
       :person/last-name
       "taateliha"}]])
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name]) ?s]
    :where '[[(lucene-text-search "(person\\/last-name:taateliha)") [[?e ?s]]]]
    :limit 1})
  (comment
    [])
  (crux/submit-tx @node [[:crux.tx/put {:crux.db/id "test"
                                        :person/last-name "test"}]])
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name])]
    :where '[[?e :person/last-name "test"]]
    :limit 1})
  (comment
    [[{:crux.db/id "test"
       :person/last-name
       "test"}]])
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name]) ?s]
    :where '[[(lucene-text-search "(person\\/last-name:test)") [[?e ?s]]]]
    :limit 1})
  (comment
    [[{:crux.db/id "test"
       :person/last-name
       "test"}
      0.13076457381248474]])
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name])]
    :where '[[?e :person/last-name "taateliha"]]
    :limit 1})
  (comment
    [[{:crux.db/id
       #uuid
        "0021cb70-4e9b-40fe-b3fc-d92ef16b02a7"
       :person/last-name
       "taateliha"}]])
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name]) ?s]
    :where '[[(lucene-text-search "(person\\/last-name:taateliha)") [[?e ?s]]]]
    :limit 1})
  (comment
    [])
```

**Comment by tkovis** [comment]
Ok it seems to be about sending ops through a tx fn.

If i redo the ops, `lucene-text-search` works
```clojure
  (crux/submit-tx
   @node
   [[:crux.tx/put (crux/entity
                   (crux/db @node)
                   #uuid "0021cb70-4e9b-40fe-b3fc-d92ef16b02a7")]])
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name]) ?s]
    :where '[[(lucene-text-search "(person\\/last-name:taateliha)") [[?e ?s]]]]
    :limit 1})
  (comment
    [[{:crux.db/id
       #uuid
        "0021cb70-4e9b-40fe-b3fc-d92ef16b02a7"
       :person/last-name
       "taateliha"}
      0.3150668740272522]])
```
If I create a dummy tx fn and use it to submit ops I can reproduce this behaviour with arbitrary docs
```clojure
  ; tx fn
  (crux/submit-tx
   @node
   [[:crux.tx/put
     {:crux.db/id :submit-tx
      :crux.db/fn '(fn [_ ops] ops)}]])

  ; submit through tx fn
  (crux/submit-tx
   @node
   [[:crux.tx/fn :submit-tx [[:crux.tx/put {:crux.db/id "test3" :person/last-name "test3"}]]]])
  
  (crux/entity (crux/db @node) "test3")
  (comment
    {:crux.db/id "test3"
     :person/last-name
     "test3"})
  
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name]) ?s]
    :where '[[(lucene-text-search "(person\\/last-name:test3)") [[?e ?s]]]]
    :limit 1})
  (comment
    [])
  
  ; redo
  (crux/submit-tx @node [[:crux.tx/put (crux/entity (crux/db @node) "test3")]])
  (crux/q
   (crux/db @node)
   {:find '[(pull ?e [:crux.db/id :person/last-name]) ?s]
    :where '[[(lucene-text-search "(person\\/last-name:test3)") [[?e ?s]]]]
    :limit 1})
  (comment
    [[{:crux.db/id "test3"
       :person/last-name
       "test3"}
      0.5472603440284729]])
```

**Comment by jarohen** [comment]
Hey @tkovis - thanks for this report, made it very easy to repro :pray: 

Have found and fixed the issue, and will include it in our `1.18.1-beta1` release later today.

James
